### PR TITLE
fix(content-manager): error when content type has no i18n enabled

### PR DESCRIPTION
**Pull request by Feranchz**
### What does it do?

Fix a bug related to the history page of content type without localisation enabled. Now we considered these type of content types and we allow saving history of content types with no i18n.

### How to test it?

1. Go to a Content Type with no i18n enabled and create and edit one entry.
2. Go to the history page of that entry
3. It should show the history without errors

### Related issue(s)/PR(s)

closes https://github.com/strapi/strapi/issues/20945

### CODE REVIEW

File: packages/core/content-manager/server/src/history/services/lifecycles.ts [hunk 1]
```diff
@@ -131,13 +131,20 @@ const createLifecyclesService = ({ strapi }: { strapi: Core.Strapi }) => {
...
           where: {
             documentId,
-            locale: { $in: locales },
-            publishedAt: null,
+            ...(isLocalizedContentType
+              ? {
+                  locale: { $in: locales },
+                  publishedAt: null,
+                }
+              : {}),
```
Comment (Marc-Roig) [inline packages/core/content-manager/server/src/history/services/lifecycles.ts:147]: the published at filter should always apply, wether the content type is localized or not

**Review by Marc-Roig — CHANGES_REQUESTED** [review]

File: packages/core/content-manager/server/src/history/services/lifecycles.ts (hunk 1 above)
Comment (Feranchz) [reply packages/core/content-manager/server/src/history/services/lifecycles.ts:147]: But entries without draft & publish are always published, so they have publishedAt with a date value (same as created I think)

Comment (Marc-Roig) [reply packages/core/content-manager/server/src/history/services/lifecycles.ts:147]: true, then that would require another check, wether the content type has D&P enabled

**Review by Marc-Roig — APPROVED** [review]
amazing! ✨

**Review by Marc-Roig — APPROVED** [review]

**Pull request merged**